osu.Game/Online/Chat/ChatLink.cs

```diff
@@ -0,0 +1,160 @@
+    {
+        private APIAccess api;
+        private BeatmapSetOverlay beatmapSetOverlay;
+        private ChatOverlay chat;
+
+        public override bool HandleInput => !string.IsNullOrEmpty(Url);
+
+        // 'protocol' -> 'https', 'http', 'osu', 'osump' etc.
+        // 'content' -> everything after '<protocol>://'
+        private Match getUrlMatch() => Regex.Match(Url, @"^(?<protocol>osu(?:mp)?|https?):\/\/(?<content>.*)");
+
+        protected override void OnLinkClicked()
+        {
+            var urlMatch = getUrlMatch();
+            if (urlMatch.Success)
+            {
+                var args = urlMatch.Groups["content"].Value.Split('/');
+
+                switch (urlMatch.Groups["protocol"].Value)
+                {
+                    case "osu":
+                        if (args.Length == 1)
+                        {
+                            base.OnLinkClicked();
+                            break;
+                        }
+
+                        switch (args[0])
+                        {
+                            case "chan":
+                                var foundChannel = chat.AvailableChannels.Find(channel => channel.Name == args[1]);
+
+                                // links should be filtered out by now if a channel doesn't exist
+                                chat.OpenChannel(foundChannel ?? throw new ArgumentException($"Unknown channel name ({args[1]})."));
+                                break;
+                            case "edit":
+                                chat.Game?.LoadEditorTimestamp();
+                                break;
+                            case "b":
+                                if (args.Length > 1 && int.TryParse(args[1], out int mapId))
+                                    beatmapSetOverlay.ShowBeatmap(mapId);
+
+                                break;
+                            case "s":
+                            case "dl":
+                                if (args.Length > 1 && int.TryParse(args[1], out int mapSetId))
+                                    beatmapSetOverlay.ShowBeatmapSet(mapSetId);
+
+                                break;
+                            case "spectate":
+                                GetUserRequest req;
+                                if (int.TryParse(args[1], out int userId))
+                                    req = new GetUserRequest(userId);
+                                else
+                                    return;
+
+                                req.Success += user =>
+                                {
+                                    chat.Game?.LoadSpectatorScreen();
+                                };
+                                api.Queue(req);
+
+                                break;
+                            default:
+                                throw new ArgumentException($"Unknown osu:// link at {nameof(ChatLink)} ({urlMatch.Groups["content"].Value}).");

```

Comment:
This will cause a hard crash. Not sure an exception is what you're looking for here.

Reply:
Oh, right. For some reason I thought that "invalid" osu:// links would be filtered out already, my bad. Just ignoring them should be fine then.